### ISSUE TITLE
Display all available 'Link' headers

### ISSUE THREAD

**Issue by rjc** [newsboat/newsboat#2060]
Some feeds have more than one link for the same entry - the main difference being the protocol, i.e.:

```
<link rel="alternate" type="text/html" "href="http://www.example.org/foo" />
<link rel="alternate" type="text/gemini" href="gemini://gemini.example.org/foo" />
<link rel="alternate" type="text/plain" href="gopher://gopher.example.org/foo" />
```

What happens now is that _last entry wins_, it seems, and the only `Link` header looks like so:

    Link: gopher://gopher.example.org/foo

I don't think _HTTP_ should automatically **win** here so, if this doesn't break anything, it would be nice to have them all displayed.

**Comment by bogdasar1985** [comment]
@rjc I have a guess why this is happening. Document parsed node by node. Each node is analyzed and information is entered into [`class Item`](https://github.com/newsboat/newsboat/blob/master/rss/item.h). Hence, in your example, the value of `Item::link` changes three times in process of parsing and, logically, stop at the last.

**Comment by bogdasar1985** [comment]
@rjc I propose the following solution. We will not display all links at once. This link is needed mainly to open the article in the browser, so when displaying the article, we will add next to the main link (say, the first one), the number of links. And when you try to open it in a browser, a window will open asking you to select a link.

**Comment by rjc** [comment]
Yes, that's correct - sometimes, when an entry is image-heavy, I want to open the link in a web browser, but other times, I might chose an alternative option.

Either way, the way I use `newsboat` is via _SSH_ without any web browser on the remote end - all I rely on is the terminal on my local machine, which ensures that all links are clickable.

I belive that, what you propose here, is to use the _built-in_ support for a web browser, in which case, I'm not sure how would that work in my scenario - using a local _GUI_ web/gemini/gopher browsers and a remote `newsboat` session via _SSH_.

**Comment by bogdasar1985** [comment]
@rjc that should work for you, because I propose that a new `ListFormAction` open with a list of links, just like that:
```
http://www.example.org/foo
gemini://gemini.example.org/foo
gopher://gopher.example.org/foo
``` 
and you can  just click in terminal.

**Comment by Minoru** [comment]
@rjc Can you give me a URL of the feed that does this? I wonder if all their links are marked with `rel="alternate"`.

At the very least, I think Newsboat should pick the link without any `rel` attributes; if there are multiple, pick `https://` or, if that's not available, `http://`. That would support the most common use-case.

To support multiple links per item, we have to extend the database (because we only have a single column for a link right now). Then, we can use the existing urls dialog to show them. I think we should still pick one link as the default, to be opened with `open`, and treat the rest as we treat links inside articles: number them and put them into the links list at the end of the article (which also makes them available through the urls dialog).

**Comment by rjc** [comment]
> Can you give me a URL of the feed that does this? I wonder if all their links are marked with rel="alternate".

Yes, [they all are](http://boston.conman.org/bostondiaries.rss).

By saying that it would be nice to have them all displayed, what I had in mind was simply not to _penalise_ non-_HTTP_ links, i.e. `gemini`, `gopher`, etc. but I guess _http(s)_ could simply be made default as they'll be the most commonly used ones.

**Comment by bogdasar1985** [comment]
@Minoru If I already stated the correct hypothesis above (I checked, everything is correct), then I think I will take up this issue.
I suggest that firstly implement a default link, and then a choice from several. The default link will not be selected in order of links follows, as it is now, but in order of priority. First `http/https`, then others.

**Comment by Minoru** [comment]
@rjc I admit I'm slow to respond right now, so might be my fault, but the provided URL doesn't contain any links with `rel=alternate`. Is there any chance this is a wrong URL?

**Comment by rjc** [comment]
@Minoru I obviously picked the wrong feed, sorry about that! [Here's the correct one](http://boston.conman.org/index.atom)! I'll edit my earlier reply in case someone reads this issue but stops there ;^)

P.S. Links to all three feeds (there's also a _JSON_ one) are available on the main [web page](http://boston.conman.org/) :^P

**Comment by Minoru** [comment]
Thanks!

I just merged the first change by @bogdasar1985, which makes Newsboat useable with such feeds by always picking an HTTP(S) link. This improves the common case. We still need a way to store and display alternative links, but this is a good start.

**Comment by bogdasar1985** [comment]
@Minoru I thought that supporting multiple links to an article was a redundant feature since this case is extremely rare. As for me, it would be good to take out the task of processing several links in a separate python script, for example. It would process the xml file with the feed, removing extra lines from it, and then newsboat would read the already processed file.
